x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/changepoint/SpikeAndDipDetectorTests.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs.changepoint;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+
+public class SpikeAndDipDetectorTests extends ESTestCase {
+
+    public void testTooLittleData() {
+        double[] values = new double[] { 1.0, 2.0, 3.0 };
+        SpikeAndDipDetector detect = new SpikeAndDipDetector(values);
+        assertThat(detect.at(0.01), instanceOf(ChangeType.Indeterminable.class));
+    }
```

Comment:
testing with an `empty` array would be good as well.

Reply:
Done